src/components/Graph.tsx

```diff
@@ -184,20 +186,40 @@ class Graph extends React.Component <Props, States> {
     };
 
     setToReplica(e: any) {
-        if (Math.abs(e.target.getStage().getPointerPosition().y - 100) < 10 && e.target.getStage().getPointerPosition().x < 700) {
-            this.setState({toReplica: 0});
+        if (e.target.getStage().getPointerPosition().x < 100) {return;}
+        if (Math.abs(e.target.getStage().getPointerPosition().y - 100) < 25 && e.target.getStage().getPointerPosition().x < 700) {
+            this.setState({toReplica: 0, toY:100});
         }
-        if (Math.abs(e.target.getStage().getPointerPosition().y - 200) < 10 && e.target.getStage().getPointerPosition().x < 700) {
-            this.setState({toReplica: 1});
+        if (Math.abs(e.target.getStage().getPointerPosition().y - 200) < 25 && e.target.getStage().getPointerPosition().x < 700) {
+            this.setState({toReplica: 1,toY:200});
         }
-        if (Math.abs(e.target.getStage().getPointerPosition().y - 300) < 10 && e.target.getStage().getPointerPosition().x < 700) {
-            this.setState({toReplica: 2});
+        if (Math.abs(e.target.getStage().getPointerPosition().y - 300) < 25 && e.target.getStage().getPointerPosition().x < 700) {
+            this.setState({toReplica: 2,toY:300});
```

Comment:
To make the code more readable and adjustable I would put all the numbers into constants. For the 3 replicas you could use an array and a loop instead of the 3 if-statements.

Reply:
I'll do it 👍